packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts

```diff
@@ -334,47 +341,50 @@ describe('paybackLoan success', () => {
     expect(vaultBefore.collateralRatio).toStrictEqual(18750) // 15000 / 80.00004568 * 100
     expect(vaultBefore.informativeRatio).toStrictEqual(18749.98929375)
 
+    const tslaInterestsPerBlockBefore = new BigNumber(netInterest * 40 / (365 * blocksPerDay)) //  netInterest * loanAmt / 365 * blocksPerDay
+    const interestsBeforePayback = await bob.rpc.loan.getInterest('scheme')
     {
-      const interests = await bob.rpc.loan.getInterest('scheme')
       const height = await bob.container.getBlockCount()
-      const tslaInterestPerBlock = new BigNumber(netInterest * 40 / (365 * blocksPerDay)).decimalPlaces(8, BigNumber.ROUND_CEIL) //  netInterest * loanAmt / 365 * blocksPerDay
-      const tslaInterestTotal = tslaInterestPerBlock.multipliedBy(height + 1 - tslaLoanHeight)
-      expect(interests[0].interestPerBlock.toFixed(8)).toStrictEqual(tslaInterestPerBlock.toFixed(8))
-      expect(interests[0].totalInterest.toFixed(8)).toStrictEqual(tslaInterestTotal.toFixed(8))
+      const tslaInterestTotal = tslaInterestsPerBlockBefore.multipliedBy(height + 1 - tslaLoanHeight)
+      expect(interestsBeforePayback[0].interestPerBlock.toFixed(8)).toStrictEqual(tslaInterestsPerBlockBefore.toFixed(8, BigNumber.ROUND_CEIL))
+      expect(interestsBeforePayback[0].totalInterest.toFixed(8)).toStrictEqual(tslaInterestTotal.toFixed(8, BigNumber.ROUND_CEIL))
     }
 
+    const payBackTslaAmount = 13
     const txid = await bob.rpc.loan.paybackLoan({
       vaultId: bobVaultId,
-      amounts: '13@TSLA',
+      amounts: `${payBackTslaAmount}@TSLA`,
       from: bobloanAddr
     })
     expect(typeof txid).toStrictEqual('string')
     await bob.generate(1)
     tslaLoanHeight = await bob.container.getBlockCount()
 
-    // assert interest by 27
+    // check the interests after payback
+    const interestAfterPayback = await bob.rpc.loan.getInterest('scheme')
+    // total interest payable is paid back upfront even at partial payback. Becuase of that actual loanDecreasedAfterPayback = (payBackTslaAmount - total interest)
+    const loanDecreasedAfterPayback = new BigNumber(payBackTslaAmount).minus(interestsBeforePayback[0].totalInterest)
+    const tslaInterestsPerBlockAfter = tslaInterestsPerBlockBefore.minus(loanDecreasedAfterPayback.multipliedBy(netInterest).dividedBy(365 * blocksPerDay))
```

Comment:
```suggestion
   const tslaInterestsPerBlockAfter = new BigNumber(40).minus(loanDecreasedAfterPayback).multipliedBy(netInterest).dividedBy(365 * blocksPerDay)
```
cpp side changed the calculations again🥲. This calculation will result in the same, But better to update as the cpp side for completeness. Hope they won't change again 
Ref - https://github.com/DeFiCh/ain/blob/207be06d35b77457b1944263b10d4be7ff09dd7e/src/masternodes/loan.cpp#L344